contracts/adapters/AaveMigrationExtension.sol

```diff
@@ -0,0 +1,615 @@
+    )
+        external
+        onlyOperator
+    {
+        _decreaseLiquidityPosition(
+            _tokenId,
+            _liquidity,
+            _amount0Min,
+            _amount1Min
+        );
+    }
+
+    /**
+     * @notice OPERATOR ONLY: Migrates a SetToken's position from an unwrapped collateral asset to another SetToken 
+     * that consists solely of Aave's wrapped collateral asset
+     * @param _decodedParams The decoded migration parameters.
+     * @param _underlyingLoanAmount The amount of unwrapped collateral asset to be borrowed via flash loan.
+     * @param _maxSubsidy The maximum amount of unwrapped collateral asset to be transferred to the Extension as a subsidy.
+     * @return underlyingOutputAmount The amount of unwrapped collateral asset returned to the operator.
+     */
+    function migrate(
+        DecodedParams memory _decodedParams,
+        uint256 _underlyingLoanAmount,
+        uint256 _maxSubsidy
+    )
+        external
+        onlyOperator
+        returns (uint256 underlyingOutputAmount)
+    {
+        // Subsidize the migration
+        if (_maxSubsidy > 0) {
+            underlyingToken.transferFrom(msg.sender, address(this), _maxSubsidy);
+        }
+
+        // Encode migration parameters for flash loan callback
+        bytes memory params = abi.encode(_decodedParams);
+
+        // Request flash loan for the underlying token
+        POOL.flashLoanSimple(
+            address(this),
+            address(underlyingToken),
+            _underlyingLoanAmount,
+            params,
+            0
+        );
+
+        // Return remaining underlying token to the operator
+        underlyingOutputAmount = _returnExcessUnderlying();
+    }
+
+    /**
+     * @dev Callback function for Aave V3 flash loan, executed post-loan. It decodes the provided parameters, conducts the migration, and repays the flash loan.
+     * @param asset The asset borrowed in the flash loan.
+     * @param amount The amount borrowed.
+     * @param premium The additional fee charged for the flash loan.
+     * @param initiator The initiator of the flash loan.
+     * @param params Encoded migration parameters.
+     * @return True if the operation is successful.
+     */
+    function executeOperation(
+        address asset,
+        uint256 amount,
+        uint256 premium,
+        address initiator,
+        bytes calldata params
+    )
+        external
+        override
+        returns (bool) 
+    {
+        require(msg.sender == address(POOL), "MigrationExtension: invalid flashloan sender");
+        require(asset == address(underlyingToken), "MigrationExtension: invalid flashloan asset");
+        require(initiator == address(this), "MigrationExtension: invalid flashloan initiator");
+
+        // Decode parameters and migrate
+        DecodedParams memory decodedParams = abi.decode(params, (DecodedParams));
+        _migrate(decodedParams);
+
+        underlyingToken.approve(address(POOL), amount + premium);
```

Comment:
I sometimes like to add a require to check if we have enough balance to repay. Not really necessary since the tx will just fail anyway if we dont, but can save a lot of time when debugging failed tx's.